### DISCUSSION TITLE
Is there an advantage to using multiple conditions in the 'Match'?

### DISCUSSION THREAD

**Discussion by don-airey** [NRules/NRules#271] [Q&A]
In my code, I often use the && operator to join conditions:

                .Match(() => security, s => fungible.FungibleId == s.FungibleId && s.IssuerId.HasValue)

But I recently noticed that your examples have additional conditions separated out as parameters to the 'Match' method:

                .Match(() => security, s => fungible.FungibleId == s.FungibleId, s => s.IssuerId.HasValue)

Is there some advantage to breaking out multiple conditions, or is the difference merely syntactic?

**Comment by a046** [comment]
Node sharing in Rete can only occur if there are common conditions across different rules. I believe NRules does this by checking the individual expressions rather than trying to decompose expressions using && operators, which means you need to use multiple conditions to facilitate node sharing (and therefore improve performance).

You could test this with the diagnostic tools and inspecting the generated graph however.

As an aside, due to the tradeoff between reuse and selectivity, NRules is also sensitive to the order of conditions when sharing nodes, so some consideration needs to be given to the order of your conditions and being consistent with the order.

**Comment by snikolayev** [comment]
@DonaldAirey I think @a046 gave a perfect answer, here. To add more color, you can also check out this SO thread: https://stackoverflow.com/questions/30570941/how-to-do-optimal-write-rule-definition-in-nrules
